app/src/features/surveys/observations/sampling-sites/edit/components/SamplingSiteEditMapControl.tsx

```diff
@@ -0,0 +1,173 @@
+import { mdiRefresh } from '@mdi/js';
+import Icon from '@mdi/react';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import IconButton from '@mui/material/IconButton';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+import { makeStyles } from '@mui/styles';
+import FileUpload from 'components/file-upload/FileUpload';
+import FileUploadItem, { IUploadHandler } from 'components/file-upload/FileUploadItem';
+import { IStaticLayer } from 'components/map/components/StaticLayers';
+import MapContainer from 'components/map/MapContainer';
+import { SurveyContext } from 'contexts/surveyContext';
+import SampleSiteFileUploadItemActionButton from 'features/surveys/observations/sampling-sites/components/SampleSiteFileUploadItemActionButton';
+import SampleSiteFileUploadItemProgressBar from 'features/surveys/observations/sampling-sites/components/SampleSiteFileUploadItemProgressBar';
+import SampleSiteFileUploadItemSubtext from 'features/surveys/observations/sampling-sites/components/SampleSiteFileUploadItemSubtext';
+import { FormikContextType } from 'formik';
+import { Feature } from 'geojson';
+import { LatLngBoundsExpression } from 'leaflet';
+import get from 'lodash-es/get';
+import { useContext, useEffect, useState } from 'react';
+import { useParams } from 'react-router';
+import {
+  calculateUpdatedMapBounds,
+  handleGPXUpload,
+  handleKMLUpload,
+  handleShapeFileUpload
+} from 'utils/mapBoundaryUploadHelpers';
+
+const useStyles = makeStyles(() => ({
+  zoomToBoundaryExtentBtn: {
+    padding: '3px',
+    borderRadius: '4px',
+    background: '#ffffff',
+    color: '#000000',
+    border: '2px solid rgba(0,0,0,0.2)',
+    backgroundClip: 'padding-box',
+    '&:hover': {
+      backgroundColor: '#eeeeee'
+    }
+  }
+}));
+
+export interface ISamplingSiteEditMapControlProps {
+  name: string;
+  mapId: string;
+  formikProps: FormikContextType<any>;
+}
+
+/**
+ * Sampling site map component.
+ *
+ * @param {ISamplingSiteEditMapControlProps} props
+ * @return {*}
+ */
+const SamplingSiteEditMapControl = (props: ISamplingSiteEditMapControlProps) => {
+  const classes = useStyles();
+  const surveyContext = useContext(SurveyContext);
+  const urlParams: Record<string, string | number | undefined> = useParams();
+  const surveySampleSiteId = Number(urlParams['survey_sample_site_id']);
```

Comment:
`Number(undefined)` will produce `NaN`. We could make `surveySampleSiteId` nullable to handle this case?
```javascript
const surveySampleSiteId: number | null = Number(urlParams['survey_sample_site_id']) || null;